qutip/core/data/dense.pyx

```diff
@@ -44,8 +44,15 @@ class OrderEfficiencyWarning(EfficiencyWarning):
 cdef class Dense(base.Data):
     def __init__(self, data, shape=None, copy=True):
         base = np.array(data, dtype=np.complex128, order='K', copy=copy)
+        # Ensure that the array is continuous.
+        # Non continuous array with copy=False would otherwise slip through
+        if not (cnp.PyArray_IS_C_CONTIGUOUS(base) or
+                cnp.PyArray_IS_F_CONTIGUOUS(base)):
+            base = base.copy()
```

Comment:
So I looked again at the `numpy.array` documentation, and it looks like this test shouldn't be necessary - instead we should set `order='A'` which is the behaviour we want.

Reply:
I did not understand the documentation the same way and upon testing:
```
>>> np.array(np.ones(4)[::2], order='A', copy=False).flags
  C_CONTIGUOUS : False
  F_CONTIGUOUS : False
...
```
So it seems `copy=False` as priority.

Reply:
Oh, you're right.  I misunderstood.  In that case, this test is right.